docs/examples/containerized_heketi_dedicated_gluster/README.md

```diff
@@ -0,0 +1,669 @@
+
+<3> copy each ssh key to the dedicated gluster cluster nodes
+
+<4> Change the permissions on the keys
+
+
+#### Obtain and edit the Heketi configuration file to setup the SSH executor.
+
+The heketi.json file is used to configure Heketi on initial container start up. There are multiple
+ways to create and have Heketi properly access and load this file, for this
+example we are simply creating the file and putting in our /usr/share/heketi working directory on each of our nodes.
+Alternatively you could create a secret and configmap to pass into the Heketi container or even store on another share
+point (GlusterFS volume or NFS, etc...).
+
+
+Below is an excerpt from the heketi.json file highlighting the sections
+that potentially could/would need to be changed for your environment.
+
+```
+	{
+  	"_port_comment": "Heketi Server Port Number",
+  	"port": "8888",  <1>
+	...
+        ...
+	"executor": "ssh",  <2>
+
+	"_sshexec_comment": "SSH username and private key file information",
+	"sshexec": {
+  	  "keyfile": "/etc/heketi/keys/heketi_key",  <3>
+  	  "user": "root",  <4>
+  	  "port": "22",  <5>
+  	  "fstab": "/etc/fstab"  <6>
+	},
+        ...
+        
+```
+<1> Default port is 8080, you can configure a different port like 8888
+
+<2> Change "executor": from mock to ssh
+
+<3> Add in the public key directory specified in previous step
+
+<4> Update SSH user (should have sudo/root type access)
+
+<5> Set the port to 22 - remove all other text
+
+<6> Set the fstab to default (etc/fstab) - remove all other text
+
+
+#### Create the dedicated GlusterFS cluster endpoints and service.
+
+The endpoints and service are used to access the GlusterFS cluster to establish mounts.
+These are used for our ReplicationController/Container in future steps and also can be reused
+for any other pods that need access to GlusterFS cluster.
+Run the following yaml examples.
+
+glusterfs-endpoints.yaml
+```
+apiVersion: v1
+kind: Endpoints
+metadata:
+ name: glusterfs-cluster
+subsets:
+ - addresses:
+   - ip: 192.168.1.205
+   ports:
+   - port: 1
+     protocol: TCP
+ - addresses:
+   - ip: 192.168.1.206
+   ports:
+   - port: 1
+     protocol: TCP
+
+```
+
+gluster-service.yaml
+```
+kind: Service
+apiVersion: v1
+metadata:
+  name: glusterfs-cluster
+spec:
+  ports:
+  - port: 1
+```
+
+
+#### Install the Heketi Container and Service.
+From the master node create the following yamls:
+
+
+heketi-service.yaml
+```
+kind: Service
+apiVersion: v1
+metadata:
+  name: heketi
+  labels:
+    name: heketi-service
+  annotations:
+    description: Exposes Heketi service
+spec:
+  type: NodePort
+  selector:
+    name: heketi
+  ports:
+  - name: heketi
+    port: 8888
+    targetPort: 8888
+
+```
+
+heketi-rc.yaml
+```
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: heketi
+spec:
+  replicas: 1
+  selector:
+    app: heketi
+  template:
+    metadata:
+      name: heketi
+      labels:
+        app: heketi
+    spec:
+      hostNetwork: true
```

Comment:
hostNetwork allowed my pod to correctly access my local private network where gluster is running (192.168.x.x).  I was not able to properly get the Heketi pod to easily communicate out of the cluster and reach my gluster nodes.  I'm not sure what I'm missing, but this seemed like a reasonable approach that assume that your nodes have network access so why not have the pod take advantage of that without having to introduce more complex network concepts (dns, routing tables, etc...).  I'm open to other suggestions but just haven't had luck without it.  This also may be a side product of running a single node local cluster as well, assuming a more production like environment would have better networking support and resources.